### PR TITLE
Feature/task has degree of kinship

### PR DESCRIPTION
**Pull request by annashamray**
Depends on https://github.com/GemeenteUtrecht/bptl/pull/45

### CODE REVIEW

File: src/bptl/work_units/brp/tests/test_task_has_degree_of_kinship_real.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+from bptl.activiti.models import ServiceTask
+
+from ..models import BRPConfig
+from ..tasks import HasDegreeOfKinship
+
+BRP_API_ROOT = "https://haalcentraal.lostlemon.nl/"
+
+
+class CreateStatusTaskTests(TestCase):
```
[the remark this answers is not in the document]
Comment (annashamray) [reply src/bptl/work_units/brp/tests/test_task_has_degree_of_kinship_real.py]: Of course, it was WIP, meaning it had a lot of garbage which I got rid of now :)

File: src/bptl/work_units/brp/tests/test_task_has_degree_of_kinship_real.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        self.assertEqual(result, {"hasDegreeOfKinship": True})
+
+    def test_kinship_4_great_great_grandchildren(self):
+        fetched_task = ServiceTask.objects.create(
+            topic_name="some-topic",
+            variables={
+                "burgerservicenummer1": "999990676",
+                "burgerservicenummer2": "999992612",
+                "kinship": 3,
```
[the remark this answers is not in the document]
Comment (annashamray) [reply src/bptl/work_units/brp/tests/test_task_has_degree_of_kinship_real.py]: Yes, it was a copypaste error and the reason why tests were failing, I fixed it

File: src/bptl/work_units/brp/tasks.py [hunk 3]
```diff
@@ -37,3 +38,49 @@ def perform(self) -> dict:
...
+        if bsn1 == bsn2:
+            return {"kinship": None}
+
+        client = get_client_class()()
+        client.task = self.task
+
+        rel1 = Relations(bsn1)
+        rel2 = Relations(bsn2)
+
+        # 1. request 1-level relations
```
Comment (joeribekker) [inline src/bptl/work_units/brp/tasks.py]: What are 1-level relations?

File: src/bptl/work_units/brp/tasks.py [hunk 4]
```diff
@@ -37,3 +38,49 @@ def perform(self) -> dict:
...
+        rel1.expand(client, 1)
+        rel2.expand(client, 1)
+
+        kinship = rel1.kinship(rel2)
+
+        if kinship:
+            return {"kinship": kinship}
+
+        # 2. Request 2-level relations
```
Comment (joeribekker) [inline src/bptl/work_units/brp/tasks.py]: If levels are tree-depths, then I would expect 4 levels (great great grand mother)? Tests seems to test for this so it probably works...

File: src/bptl/work_units/brp/tasks.py [hunk 5]
```diff
@@ -37,3 +38,49 @@ def perform(self) -> dict:
...
+
+    def perform(self) -> dict:
+        variables = self.task.get_variables()
+        bsn1 = variables["burgerservicenummer1"]
+        bsn2 = variables["burgerservicenummer2"]
+
+        if bsn1 == bsn2:
+            return {"kinship": None}
+
+        client = get_client_class()()
```
Comment (joeribekker) [inline src/bptl/work_units/brp/tasks.py]: Is this a special client that I don't know about but talks json+hal? :)

**Comment by annashamray** [comment]
> It looks good overall but I have a hard time understanding the code exactly...
> 
> In general, could you add some more comments (on functions and as comments between code lines) on what it does?

@joeribekker additional comments and docstrings are added

File: src/bptl/work_units/brp/tasks.py [hunk 6]
```diff
@@ -37,3 +38,52 @@ def perform(self) -> dict:
         is_above_age = None if leeftijd is None else leeftijd >= age
 
         return {"isAboveAge": is_above_age}
+
+
+@register
+class DegreeOfKinship(WorkUnit):
+    """
+    Fetches BRP API and returns flag if certain two people have a specified degree of kinship
```
Comment (sergei-maertens) [inline src/bptl/work_units/brp/tasks.py]:
```suggestion
    Retrieve the degree of kinship from the BRP API.
```

File: src/bptl/work_units/brp/tasks.py [hunk 7]
```diff
@@ -37,3 +38,52 @@ def perform(self) -> dict:
...
+
+    Required process variables:
+
+    * ``burgerservicenummer1``: BSN of the first person
+    * ``burgerservicenummer2``: BSN of the second person
+
+    The task sets the process variables:
```
Comment (sergei-maertens) [inline src/bptl/work_units/brp/tasks.py]:
```suggestion
    **Sets the process variables**
```

File: src/bptl/work_units/brp/tasks.py [hunk 8]
```diff
@@ -37,3 +38,52 @@ def perform(self) -> dict:
...
+
+    * ``kinship``: integer, which represents the degree of kinship (blood relations). Values can be in range [1..4]
```
Comment (sergei-maertens) [inline src/bptl/work_units/brp/tasks.py]:
```suggestion
    * ``kinship``: integer, which represents the degree of kinship (blood relations). Values can be in 
       range [1..4] or ``Null`` if the BSNs are identical.
```

File: src/bptl/work_units/brp/tasks.py (hunk 5 above)
Comment (sergei-maertens) [reply src/bptl/work_units/brp/tasks.py]: We'll do some cleanup w/r to the client class stuff later, especially with the pendin zgw-consumers rework.

**Pull request merged**